src/main/java/com/couchbase/lite/store/ForestDBStore.java

```diff
@@ -1047,24 +1047,20 @@ public Status run() {
     }
 
     @Override
-    public ViewStore getViewStorage(String name, boolean create) {
-        try {
-            return new ForestDBViewStore(this, name, create);
-        } catch (CouchbaseLiteException e) {
-            if(e.getCBLStatus().getCode() != Status.NOT_FOUND)
-                Log.e(TAG, "Error in getViewStorage()", e);
-            return null;
-        }
+    public ViewStore getViewStorage(String name, boolean create) throws CouchbaseLiteException {
+        return new ForestDBViewStore(this, name, create);
     }
 
     @Override
     public List<String> getAllViewNames() {
         List<String> result = new ArrayList<String>();
         String[] fileNames = new File(directory).list();
-        for(String filename : fileNames){
-            String viewName = ForestDBViewStore.fileNameToViewName(filename);
-            if(viewName != null)
-                result.add(viewName);
+        for (String filename : fileNames) {
+            try {
+                result.add(ForestDBViewStore.fileNameToViewName(filename));
+            } catch (CouchbaseLiteException e) {
+                Log.w(TAG, "Error in fileNameToViewName(): filename=" + filename, e);
```

Comment:
Do you intend to put Log.w? Why isn't it Log.e? There are a few cases like this below as well?